File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/events/IEvent.java
package com.mapbox.rctmgl.events;

import com.facebook.react.bridge.WritableMap;

/**
 * Created by nickitaliano on 8/23/17.
 */

public interface IEvent {
    int getID();
    String getKey();
    String getType();
    long getTimestamp();
    boolean equals(IEvent event);
    boolean canCoalesce();
    WritableMap getPayload();
    WritableMap toJSON();
}

File: android/src/main/java/com/rnmapbox/rnmbx/events/IEvent.kt
package com.rnmapbox.rnmbx.events

import com.facebook.react.bridge.WritableMap

/**
 * Created by nickitaliano on 8/23/17.
 */
interface IEvent {
    val iD: Int
    val key: String?
    val type: String?
    val timestamp: Long
    fun equals(event: IEvent): Boolean
    fun canCoalesce(): Boolean
    val payload: WritableMap
    fun toJSON(): WritableMap
}